### PR TITLE
Add Haskell Language Server check to CI

### PR DESCRIPTION
**Pull request by palas**
This PR adds a check to the CI that ensures that it works with the developer shell as it is.

# Changelog

```yaml
- description: |
    Add CI check that ensures HLS works with developer shell
  type:
  - test
```

# Context

Recently we had an issue with HLS not working because of a bug. We could have catched that if we had a test in the CI that ensured HLS worked. This PR aims to address this.

# How to trust this PR

Check the workflow makes sense. But basically, I would just look at [the CI's output](https://github.com/IntersectMBO/cardano-cli/actions/runs/9618399739/job/26532205253?pr=798).

# Checklist

- [x] Commit sequence broadly makes sense and commits have useful messages
- [x] New tests are added if needed and existing tests are updated. See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [x] Self-reviewed the diff

### CODE REVIEW

File: .github/workflows/hls.yml
Comment (smelc) [inline .github/workflows/hls.yml]:
```suggestion
      - uses: actions/checkout@v4
      - name: Install Nix with good defaults
        uses: input-output-hk/install-nix-action@v20
        with:
          extra_nix_config: |
            trusted-public-keys = cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY= hydra.iohk.io:f/Ea+s+dFdN+3Y/G+FDgSq+a5NEWhJGzdjvKNGv0/EQ=
            substituters = https://cache.iog.io/ https://cache.nixos.org/
          nix_path: nixpkgs=channel:nixos-unstable
      - uses: cachix/install-nix-action@v18
        with:
          nix_path: nixpkgs=channel:nixos-unstable
      # Make the Nix environment available to next steps
      - uses: rrbutani/use-nix-shell-action@v1
      - run: haskell-language-server
```

Rationale: use same setup as other workflows

Comment (palas) [reply .github/workflows/hls.yml]: Solved in: https://github.com/IntersectMBO/cardano-cli/pull/798/commits/05d4c08d687cb54423fec424a66f0db3312501d7
I had to add `cabal update` to the `run` bit too.

**Review by smelc — APPROVED** [review]

**Review by carbolymer — CHANGES_REQUESTED** [review]
I'm fine with the version bump. This GHA however takes 20 minutes, which is too long of a wait for each PR to get merged.

I don't remember, can you confirm that each PR will wait for this new check to complete before merging?

**Comment by smelc** [comment]
@carbolymer> right now the new check is not _required_ so we can merge manually before this new check is finished.

Regarding the merge queue's behavior, [the documentation](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-a-merge-queue) says it waits for required checks to be satisfied. So not sure if it merges as soon as _required_ checks are green, ignoring the others. I'd be surprised if it was the case.

File: .github/workflows/hls.yml
Comment (smelc) [inline .github/workflows/hls.yml]:
```suggestion
          echo "VERSION=$(ghc --numeric-version)" >> $GITHUB_OUTPUT
```

File: .github/workflows/hls.yml
Comment (smelc) [inline .github/workflows/hls.yml]: Also maybe include a mechanism similar to `CABAL_CACHE_VERSION` from the CLI? I'm pretty confident at some point the cache will get wrong/corrupted :upside_down_face:

File: .github/workflows/hls.yml
Comment (smelc) [inline .github/workflows/hls.yml]:
```suggestion
        uses: actions/cache@v4
```

Why not use the latest version available?

File: .github/workflows/hls.yml
Comment (palas) [reply .github/workflows/hls.yml]: Oh, that is convenient

File: .github/workflows/hls.yml
Comment (palas) [reply .github/workflows/hls.yml]: I was worrying about the same, I will ~include the same variable here~ do the same thing here

**Comment by palas** [comment]
> I'm fine with the version bump. This GHA however takes 20 minutes, which is too long of a wait for each PR to get merged.
> 
> I don't remember, can you confirm that each PR will wait for this new check to complete before merging?

It takes 4 min now on cache hit. And, it is my understanding, that it should hit cache as long as dependencies for the project don't change

hls-cached

**Review by carbolymer — APPROVED** [review]
If it takes 4 mins, I'm fine with it! 🚀

**Pull request merged**
